src/docs/release/breaking-changes/hero-controller-scope.md

```diff
@@ -0,0 +1,139 @@
+---
+title: More Strict Assertions in the Navigator and the Hero Controller Scope
+description: Added additional assertions to guarantee that one hero controller scope can only subscribe to one navigator at a time.
+---
+
+## Summary
+
+The framework throws assertion error when it detects there are
+multiple navigators register with one hero controller scope.
+
+## Context
+
+The hero controller scope hosts a hero controller for its widget
+subtree. The hero controller can only support one navigator at
+a time. Previously, there was no assertion to guarantee that.
+
+## Description of change
+
+If the code starts throwing assertion error after this change,
+it means the code has already been broken even before this change.
+This change only surfaced the problem.
+
+
+## Migration guide
+
+An example application that starts to throw exceptions.
+
+```dart
+import 'package:flutter/material.dart';
+
+void main() {
+  runApp(
+    MaterialApp(
+      builder: (BuildContext context, Widget child) {
+        // Builds two parallel navigators. This will throw
+        // error because both of navigators are under the same
+        // hero controller scope created by MaterialApp.
+        return Stack(
+          children: <Widget>[
+            Navigator(
+              onGenerateRoute: (RouteSettings settings) {
+                return MaterialPageRoute<void>(
+                  settings: settings,
+                  builder: (BuildContext context) {
+                    return const Text('first Navigator');
+                  }
+                );
+              },
+            ),
+            Navigator(
+              onGenerateRoute: (RouteSettings settings) {
+                return MaterialPageRoute<void>(
+                  settings: settings,
+                  builder: (BuildContext context) {
+                    return const Text('Second Navigator');
+                  }
+                );
+              },
+            ),
+          ],
+        );
+      }
+    )
+  );
+}
+```
+
+You can fix this application by introducing your own hero controller scopes.
+
+```dart
```

Comment:
@chunhtai, this code is breaking the Travis build because HeroController is not yet in the stable release. Can you mark this with the `<!--skip-->` tag?